src/test/java/com/vinylteam/vinyl/util/RawOffersSorterTest.java

```diff
@@ -242,6 +247,24 @@ void getParametersForComparisonWhenStringContainsManyWordsWithoutArticleTest() {
         assertEquals("best", parameterForComparison);
     }
 
+    static Stream<Arguments> getTestData() {
+        return Stream.of(
+                Arguments.of(3, new String[]{"hello", "variOus", "artist"}, "Hello Vinyl Artist Various"),
+                Arguments.of(0, new String[]{"hello12", "variOus12", "artist12"}, "Hello Vinyl Artist Various"),
+                Arguments.of(1, new String[]{"variOus"}, "Hello Vinyl Artist Various "),
+                Arguments.of(1, new String[]{"variOus", "Various"}, "Hello Vinyl Artist Various "),
+                Arguments.of(1, new String[]{"  variOus ", "  Various    "}, "Hello Vinyl Artist Various ")
+        );
+    }
+
+    @ParameterizedTest(name = "#{index} - Run test with args={0}, {1}, {2}")
+    @MethodSource("getTestData")
+    @DisplayName("Checks whether match counter works correctly")
+    void getMatchNumberTest(int expectedNumber, String[] searchWords, String rawOfferFullName) {
+        int number = sorter.getMatchNumber(searchWords, rawOfferFullName);
```

Comment:
getMatchingRate?

Reply:
Shouldn't expectedNumbers be set within the test?
number - actualNumber, but I'm just cherrypicking.

Reply:
getMatchingRate - good name, agreed